tests/action.test.ts

```diff
@@ -245,6 +247,125 @@ describe('Integration tests - action.ts', () => {
     expect(mocks.TFError).not.toHaveBeenCalled();
   });
 
+  test('Hardware test', async () => {
+    setDefaultInputs();
+    // Mock required Action inputs
+    // api_key - A testing farm server api key
+    vi.stubEnv('INPUT_API_KEY', 'abcdef-123456');
+    // git_url - An url to the GIT repository
+    vi.stubEnv(
+      'INPUT_GIT_URL',
+      'https://github.com/sclorg/testing-farm-as-github-action'
+    );
+    // tmt_plan_regex - A tmt plan regex which will be used for selecting plans. By default all plans are selected
+    vi.stubEnv('INPUT_TMT_PLAN_REGEX', 'fedora');
+    vi.stubEnv('INPUT_TMT_HARDWARE', '{"memory": ">= 8 GB"}');
+
+    // Mock Testing Farm API
+    vi.mocked(mocks.unsafeNewRequest).mockImplementation(
+      async (_request: unknown) => {
+        return Promise.resolve({
+          id: '1',
+        });
+      }
+    );
+    vi.mocked(mocks.requestDetails)
+      .mockResolvedValueOnce({ state: 'new', result: null })
+      .mockResolvedValueOnce({ state: 'queued', result: null })
+      .mockResolvedValueOnce({ state: 'pending', result: null })
+      .mockResolvedValueOnce({ state: 'running', result: null })
+      .mockResolvedValueOnce({
+        state: 'complete',
+        result: { overall: 'passed', summary: '\\o/' },
+      });
+
+    // Run action
+    const octokit = new Octokit({ auth: 'mock-token' });
+    const pr = await PullRequest.initialize(1, octokit);
+
+    await action(pr);
+
```

Comment:
Maybe also test that `newReqeust()` wasn't called?

Reply:
Fixed by https://github.com/sclorg/testing-farm-as-github-action/pull/146/commits/fb43f94db0bde185de3d4635938e3c90baf6851f